switchdev_utils/config-service/switchdev_network_config

```diff
@@ -0,0 +1,678 @@
+                     % (sriov_pf, str(e)))
+
+
+def wait_for_uplink_rep_creation(sriov_pf):
+    uplink_rep_phys_switch_id_path = "/sys/class/net/%s/phys_switch_id" \
+                                     % sriov_pf
+
+    for i in range(MAX_RETRIES):
+        if get_file_data(uplink_rep_phys_switch_id_path):
+            logger.info("Uplink representor %s ready", sriov_pf)
+            break
+        time.sleep(1)
+    else:
+        raise RuntimeError("Timeout while waiting for uplink representor %s.",
+                           sriov_pf)
+
+
+def configure_switchdev(sriov_pf):
+    logger.info("Configuring switchdev for %s" % sriov_pf)
+    pf_pci = get_pf_pci(sriov_pf)
+    pf_device_id = get_pf_device_id(sriov_pf)
+    if pf_device_id == CX4_DEVICE_ID or pf_device_id == CX4LX_DEVICE_ID:
+        try:
+            run_command('/usr/sbin/devlink', 'dev', 'eswitch', 'set',
+                        'pci/%s' % pf_pci, 'inline-mode', 'transport')
+        except Exception as e:
+            logger.error("Failed to set inline-mode to transport%s" % str(e))
+            raise e
+    try:
+        run_command('/usr/sbin/devlink', 'dev', 'eswitch', 'set',
+                    'pci/%s' % pf_pci, 'mode', 'switchdev')
+    except Exception as e:
+        logger.error("Failed to set mode to switchdev for interface %s\n%s"
+                     % (sriov_pf, str(e)))
+        raise e
+    logger.info("Device pci/%s set to switchdev mode." % pf_pci)
+    wait_for_uplink_rep_creation(sriov_pf)
+
+
+def reload_udev_rules():
+    try:
+        run_command('/usr/sbin/udevadm', 'control', '--reload-rules')
+        logger.info("udev rules reloaded successfully")
+    except Exception as e:
+        logger.error("Failed to reload udev rules \n%s" % str(e))
+        raise e
+
+
+def write_config(filename, data, append=False):
+    if append:
+        with open(filename, 'a') as f:
+            f.write(str(data))
+    else:
+        with open(filename, 'w') as f:
+            f.write(str(data))
+
+
+def add_udev_rule(udev_data, udev_file):
+    trigger_udev_rule = False
+    udev_data = udev_data.strip()
+    if not os.path.exists(udev_file):
+        write_config(udev_file, "%s\n" % udev_data)
+        reload_udev_rules()
+        trigger_udev_rule = True
+    else:
+        file_data = get_file_data(udev_file)
+        udev_lines = file_data.split("\n")
+        if udev_data not in udev_lines:
+            write_config(udev_file, "%s\n" % udev_data, append=True)
+            reload_udev_rules()
+            trigger_udev_rule = True
+    return trigger_udev_rule
+
+
+def create_rep_link_name_script():
+    with open(REP_LINK_NAME_FILE, "w") as f:
+        f.write(REP_LINK_NAME_DATA)
+    # Make the _REP_LINK_NAME_FILE executable
+    os.chmod(REP_LINK_NAME_FILE, 0o755)
+
+
+def add_udev_rule_to_unmanage_vf_representors_by_nm():
+    udev_data_line = 'SUBSYSTEM=="net", ACTION=="add", ATTR{phys_switch_id}'\
```

Comment:
This will unmanage all vf representors, regardless if their PF was mentioned in the config file.

is that how os-net-config work ?